src/Cassandra/IMetric.cs

```diff
@@ -0,0 +1,28 @@
+﻿// 
+//       Copyright (C) DataStax Inc.
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//       http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+namespace Cassandra
```

Comment:
I've noticed that some part of the API is on the root namespace, is this intentional?

I think we can expose them in `Metrics` namespace, like `IDriverMetrics` and `IMetricsRegistry`.

Reply:
I tried to have everything related to the `Builder` on the root namespace so that when users try to call `builder.WithMetrics(root.CreateDriverMetricsProvider(), new MetricOptions())`, they wouldn't run into the namespace issue. This means that `NodeMetric, `SessionMetric`, `MetricOptions` and the extension `CreateDriverMetricsProvider` should be on the root namepace. Maybe I should move the other extension methods to the same namespace as `IDriverGauge`, etc.

Reply:
I think it could be cleaner to move everything to `Metrics` namespace.
There are currently a lot of public members in the root namespace for historical reasons, but we can use correct ones for new features.

Reply:
:+1: moved `SessionMetric, `IMetric`, `NodeMetric` and `MetricOptions` to `Cassandra.Metrics`